src/types/metadata.ts

```diff
@@ -12,6 +12,7 @@ export type BaseDefinition = {
   plans_using: Pick<Plan, 'id'>[];
   revision: number;
   tags: { tag: Tag }[];
+  type: "JAR" | "EDSL";
```

Comment:
does this `type` apply to all other definition types (i.e. constraints, scheduling conditions) as well?

Reply:
The goal is eventually to have procedural constraints and conditions, but currently we just support procedural scheduling goals. I've made the change to just add this field to `SchedulingGoalDefinition` for now.